saleor/graphql/account/mutations/account.py

```diff
@@ -253,13 +254,16 @@ def check_permissions(cls, context):
 
     @classmethod
     def perform_mutation(cls, root, info, **data):
+        address_data = data["input"]
         success_response = super().perform_mutation(root, info, **data)
         address_type = data.get("type", None)
         user = info.context.user
         success_response.user = user
         if address_type:
-            instance = success_response.address
-            utils.change_user_default_address(user, instance, address_type)
+            address = cls.validate_address(
```

Comment:
This validation is executed after the address update inside `perform_mutation`.  Do we have a negative test to verify the address not change after validation fails?

Reply:
Fixed